Gems/Atom/RPI/Code/Source/RPI.Public/Pass/Pass.cpp

```diff
@@ -113,6 +117,26 @@ namespace AZ
             return desc;
         }
 
+        int Pass::GetDeviceIndex() const
+        {
+            AZ_Assert(m_state != PassState::Uninitialized && m_state != PassState::Reset,
+                "Getting device index before building might return a wrong result since its parent pass is not yet ready.");
+            if (m_deviceIndex == AZ::RHI::MultiDevice::InvalidDeviceIndex && m_parent)
+            {
+                return m_flags.m_parentDeviceIndexCached ? m_parentDeviceIndex : m_parent->GetDeviceIndex();
+            }
+            return m_deviceIndex;
+        }
+
+        int Pass::RecursiveGetDeviceIndex() const
+        {
+            if (m_deviceIndex == AZ::RHI::MultiDevice::InvalidDeviceIndex && m_parent)
+            {
+                return m_parent->RecursiveGetDeviceIndex();
+            }
+            return m_deviceIndex;
+        }
```

Comment:
Does this really need to be two methods?
The only difference being that the first might take the cached if available, but other than that both are "recursive".